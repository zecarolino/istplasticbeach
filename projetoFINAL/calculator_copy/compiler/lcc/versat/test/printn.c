void putchar(int ch) { asm("\trdw R11\n\taddi 2\n\twrw RB\n\trdwb\n\trdwb\n\twrw CPRT_BASE\n"); }
int _mod(int a, int b) {
	while (a >= b) a -= b;
	return a;
}
int _div(int a, int b) {
	int x = 0;
	while (a >= b) a -= b, x++;
	return x;
}
void printn(int n, int b) {
	int a;
	if (a = n / b) printn(a, b);
	putchar(n % b + '0');
}
int main() { printn(579, 10); putchar('\n'); return 0; }
/* 13579 => 136015 cycles
 * 3579 => 36754 cycles
 * 579 => 6735 cycles
 */
