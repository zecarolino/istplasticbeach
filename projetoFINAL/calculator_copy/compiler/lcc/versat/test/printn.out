VCD info: dumpfile xtop.vcd opened for output.
579
0
Catched Trap at time  7092 clock cycles:
